squidpy/datasets/_10x_datasets.py

```diff
@@ -0,0 +1,140 @@
+from typing import Literal, Optional
+from pathlib import Path
+from collections import namedtuple
+
+from scanpy import _utils
+from anndata import AnnData
+from scanpy._settings import settings
+from scanpy.readwrite import read_visium
+
+from squidpy._constants._constants import TenxVersions
+
+VisiumFiles = namedtuple("VisiumFiles", ["feature_matrix", "spatial_attrs", "tif_image"])
+
+_VisiumDatasets = Literal[
```

Comment:
yes indeed, I was afraid of that, so you'd suggest to just use string and then format options in `_docs`? But we'd still ahve to check that it's "typed" correctly? what about a constant?

also @dineshpalli random note but remember that in docs we need to put reference to link

Reply:
> yes indeed, I was afraid of that, so you'd suggest to just use string and then format options in _docs? But we'd still ahve to check that it's "typed" correctly? what about a constant?

This is a tough one. The cleanest way would be to remove `TenxVersions`, since the version is determines purely from the passed sample id string and define a new enum as:
```python3
from enum import auto
TenxVersion = Literal["1.1.0", "1.2.0", "1.3.0"]

class TenxDatsets(str, ModeEnum):
    V1_Breast_Cancer_Block_A_Section_1 = auto()
    ...

    @staticmethod
    def _generate_next_value_(name: str, start: int, count: int, last_values: List[Any]) -> str:
        # we should include name.lower() in `ModeEnum` later to save ourselves typing
        return name

    @property
    def version(self) -> TenxVersion:
        if self.value.startswith("V_1"):
            return "1.1.0"
        ...
    
    @property
    def metadata(self) -> VisiumFiles:
        return VisiumFiles(...)
```

Of course the fundamental problem is what happens when a new version comes in, since we're locked into the current 3 ones. This could be solved by exposing version as
```python3
def visium(sample_id: TenxDatasets | str,
           version: Optional[TenxVersion] = None,
           ...) -> AnnData:
    if version is None:
         file_metadata = TenxDatasets(samle_id).metadata
    else:
         # new version which we haven't handled in `TenxDatasets`
         file_metadata = VisiumFiles(...)
    ...
```

Reply:
I like it, actually not too much of a problem when new version comes in we can also modify the `Literal` right?

Reply:
Think we can postpone this for now.